notification-portlet-webapp/src/main/java/org/jasig/portlet/notice/service/jpa/JpaNotificationDao.java

```diff
@@ -0,0 +1,119 @@
+/**
+ * Licensed to Jasig under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work
+ * for additional information regarding copyright ownership.
+ * Jasig licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file
+ * except in compliance with the License. You may obtain a
+ * copy of the License at:
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.jasig.portlet.notice.service.jpa;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import javax.persistence.EntityManager;
+import javax.persistence.PersistenceContext;
+import javax.persistence.TypedQuery;
+
+import org.apache.commons.lang.Validate;
+import org.jasig.portlet.notice.NotificationState;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Repository;
+import org.springframework.transaction.annotation.Transactional;
+
+/**
+ * This DAO class handles raw CRUD operations for the JPA-flavor notifications.
+ *
+ * @author drewwills
+ */
+@Repository
+/* package-private */ class JpaNotificationDao implements INotificationDao {
+
+    @PersistenceContext
+    private EntityManager entityManager;
+
+    private final Logger log = LoggerFactory.getLogger(getClass());
+
+    /**
+     * Search for a JpaEntry with the specified Id. If the entry exists in the
+     * persistence context, it is returned;  otherwise, null is returned.
+     */
+    @Override
+    public JpaEntry getEntry(long entryId) {
```

Comment:
I discovered during testing that Postgres has an issue with the fact that some of these methods do not have @Transactional on them. Without that, I would get errors like: 

Caused by: org.postgresql.util.PSQLException: Large Objects may not be used in auto-commit mode.

(which makes little sense as we are not using clobs or blobs, but that isn't the point).